DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs

```diff
@@ -169,46 +179,110 @@ public void UnlockAccount_unlocks_account_and_returns_to_page()
             var result = administratorController.UnlockAccount(1);
 
             // Then
-            A.CallTo(() => userDataService.UpdateAdminUserFailedLoginCount(1, 0)).MustHaveHappened();
-            result.Should().BeRedirectToActionResult().WithActionName("Index");
+            using (new AssertionScope())
+            {
+                A.CallTo(() => userDataService.UpdateAdminUserFailedLoginCount(1, 0)).MustHaveHappened();
+                result.Should().BeRedirectToActionResult().WithActionName("Index");
+            }
+        }
+
+        [Test]
+        public void DeactivateAdminUser_returns_not_found_when_trying_to_access_page_for_own_admin_account()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser();
+            var loggedInAdminUser = adminUser;
+
+            A.CallTo(() => userDataService.GetAdminUserById(adminUser.Id)).Returns(adminUser);
+            A.CallTo(() => userDataService.GetAdminUserById(loggedInAdminUser.Id)).Returns(loggedInAdminUser);

```

Comment:
Doesn't this just duplicate the previous line?

Reply:
Different IDs return different users. Needed for getting past the checks to ensure a user does not deactivate themselves.

Reply:
Ah right. Think I was getting confused with a different test. Looks good to me now.